### ISSUE TITLE
Units movement wrong during side turn events in replays causes OOS in 1.12 (GNA #23314)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1452]
**[Original submission](https://gna.org/bugs/index.php?23314)** by anonymous on 2015-02-23

Put this event in an scenario while in the normal game it will only fire if there is a unit that didn't move last turn, it will fire every side turn in a replay.

The reason is that units movement is reset to soon (before the side turn event) in the replay (in this line: https://github.com/wesnoth/wesnoth/blob/0da5b24a76103428181b555bd38c2b6976895db7/src/replay_controller.cpp#L621)

```

	[event]

		first_time_only = no

		name = "side turn"

		[filter_condition]

			[have_unit]

				side = "$side_number"

				formula # "movement_left  total_movement"

			[/have_unit]

		[/filter_condition]

		[message]

			message = "there is a unit with full movment left"

			speaker = "narrator"

		[/message]

	[/event]

```

(Reproduced on all)
Release: 1.12.0
Priority: 5 - Normal
Severity: 4 - Important

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-10-15**

**gfgtdf changed release:** 1.12.0, 1.13-dev -> 1.12.0

**gfgtdf wrote:**

> This commit https://github.com/wesnoth/wesnoth/commit/bc846c73acc459abe5e87b2101c7298d6dc88e1c has fixed this in wesnoth 1.13 . So this bug should onyl effect wesnoth 1.12 now.

**Comment by wesnoth-bugs** [comment]
**Modified on 2016-10-15**

**gfgtdf wrote:**

> Actualyl that commit message was wrong and that commit fixed a different bug (https://gna.org/bugs/?23316). still this bug is fixed in master and onyl present on 1.12

**Comment by wesnoth-bugs** [comment]
**Modified on 2017-03-26**

**gfgtdf changed title:** Units movement wrong during side turn events in replays causes OOS -> Units movement wrong during side turn events in replays causes OOS in 1.12
